### PR TITLE
HOCS-4440: WebForm complaints allows early closure

### PR DESCRIPTION
**Pull request by LiamMacP**
This changes support the requirement for a user to be able to close a
WebForm case as early as possible if it's not deemed valid. This is done
through a check if the channel is specified as WebForm and then shows a
validation screen within COMP_REGISTRATION. The outcome of this screen
is then propagated up to the COMP workflow to complete the case as
required.

WIP change for closing Webform cases earlier in the pipeline. This
contains the core logic that works - the other parts (form design)
should be easily alterable).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -149,21 +149,7 @@ task createCOMPSnapshots {
 
 task checkCOMPSnapshots {
 
-    doFirst {
```
Comment (robertdeniszczyc2) [inline build.gradle:152]: Is this removal intentional?

File: src/main/java/uk/gov/digital/ho/hocs/workflow/api/WorkflowService.java [hunk 2]
```diff
@@ -103,18 +103,18 @@ public CreateCaseResponse createCase(String caseDataType, LocalDate dateReceived
...
                 seedData.put(WorkflowConstants.TOPICS, receivedData.get(WorkflowConstants.TOPICS));
                 seedData.put(WorkflowConstants.REQUEST_QUESTION, receivedData.get(WorkflowConstants.REQUEST_QUESTION));
                 seedData.put(WorkflowConstants.ORIGINAL_CHANNEL, receivedData.get(WorkflowConstants.ORIGINAL_CHANNEL));
-            } else {
-                seedData.put(WorkflowConstants.CASE_REFERENCE, caseResponse.getReference());
-                seedData.put(WorkflowConstants.DATE_RECEIVED, dateReceived.toString());
-                seedData.put(WorkflowConstants.LAST_UPDATED_BY_USER, userUUID.toString());
+            } else if (Objects.equals(caseDataType, WorkflowConstants.CASE_DATA_TYPE_COMP)
+                    && receivedData != null
+                    && Objects.equals(receivedData.get(WorkflowConstants.CHANNEL), WorkflowConstants.CHANNEL_COMP_WEBFORM)) {
```
Comment (robertdeniszczyc2) [inline src/main/java/uk/gov/digital/ho/hocs/workflow/api/WorkflowService.java:112]: Is it worth extracting this logic chain into a method with a more readable name so it's easier to understand this block? Would also be able to add test for the logic chain easily then too.

**Review by robertdeniszczyc2 — CHANGES_REQUESTED** [review]
Added a couple questions,

Cheers,
Rob

File: build.gradle (hunk 1 above)
Comment (LiamMacP) [reply build.gradle:152]: Not intentional - was just for initial testing - added back

File: src/main/java/uk/gov/digital/ho/hocs/workflow/api/WorkflowService.java (hunk 2 above)
Comment (LiamMacP) [reply src/main/java/uk/gov/digital/ho/hocs/workflow/api/WorkflowService.java:112]: Good shout simplified it in this area - and added a test

**Review by robertdeniszczyc2 — APPROVED** [review]
